src/Components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */
+
+import './Dropdown.scss';
+import React, { useCallback, useState } from 'react';
+import { debounce } from 'lodash';
+import classNames from 'classnames';
+
+import { Person } from '../../types/Person';
+import { peopleFromServer } from '../../data/people';
+
+type Props = {
+  delay: number,
+  setSelectedPerson: (person: Person | null) => void,
+};
+
+export const Dropdown: React.FC<Props> = ({ delay, setSelectedPerson }) => {
+  const [query, setQuery] = useState('');
+  const [completeQuery, setCompleteQuery] = useState('');
+
+  const normalizeQuery = useCallback((someQuery: string) => {
+    return someQuery.toLowerCase().trim();
+  }, []);
+
+  const filterPeople = useCallback(
+    (peopleArray: Person[], complQuery: string) => {
+      return peopleArray.filter(person => {
+        return normalizeQuery(person.name)
+          .includes(normalizeQuery(complQuery));
+      });
+    }, [query],
+  );
+  // eslint-disable-next-line
+  const filteredPeople: Person[] = filterPeople(peopleFromServer, completeQuery);
```

Comment:
try to use `useMemo` hook
```suggestion
  const filteredPeople = useMemo(...)
```